CONTRIBUTING.md

```diff
@@ -0,0 +1,178 @@
+
+```bash
+Copy code
+git clone https://github.com/your-username/project.git
+```
+
+Replace "your-username" with your GitHub username. 
+
+#### Create a branch
+Create a new branch to work on your changes using the following command:
+
+```bash
+git checkout -b my-feature
+```
+
+Replace "my-feature" with a descriptive name for your feature or fix.
+
+#### Make changes
+Make the necessary changes to the code, documentation, or other project files.
+
+#### Commit changes
+Once you've made your changes, commit them to your local repository using the following commands:
+
+```bash
+git add . # or be explicit and add only the changes you really want to push
+git commit -m "Add new feature"  # Replace with a descriptive commit message
+```
+
+#### Push changes
+Push your changes to your GitHub repository:
+
+```bash
+git push origin my-feature
+```
+
+#### Submit a pull request
+Go to your fork of the repository on GitHub (or to the original repository if you did not fork it) and click the 
+"New Pull Request" button. Compare the changes you made in your branch with the main branch of the original repository.
+Provide a title and description for your pull request, detailing the changes you made. Submit your changes for review.
+
+#### Review and iterate
+A project maintainer will review your pull request and provide feedback. Make any requested changes and push them to 
+your branch. The pull request will be updated automatically.
+
+#### Merge pull request
+Once your pull request is approved, a project maintainer will merge your changes into the main branch of the repository.
+
+### Code style
+This code adheres to the [PEP 8 style guide](https://peps.python.org/pep-0008/), which promotes readability and 
+consistency in Python code:
+
+- Functions are defined using snake_case.
+- Comments start with # and are used to explain code functionality.
+- Docstrings (multiline strings enclosed in triple quotes) are used to document non-private functions or methods.
+- Indentation is consistent and consists of four spaces per level.
+- if `__name__ == "__main__":` block is used to allow the script to be executed directly as well as imported as a module.
+- private functions are defined using `_name_of_private_function`.
+
+### Testing
+The project uses the unittest framework in Python. We define a function e.g., `add(a, b)` that adds two numbers.
+We create a test class `TestAddFunction` that inherits from unittest.TestCase. Inside the test class, we define 
+multiple test methods, each testing a specific scenario of the add function. Each test method starts with the word test 
+to indicate that it's a test case. Inside each test method, we call the function being tested with specific inputs 
+and use assert methods like `assertEqual` to verify the expected behavior. In each test method we use three conceptual
+blocks to help reading and understanding the test: `# Given`; `# When`; and `# Then`. 
```

Comment:
Some details on `# Given`; `# When`; and `# Then` in listed points?

Reply:
I don't want to make this file too long. I think - for some things - the user can see examples in the code.